test/unit/parser_spec.js

```diff
@@ -151,17 +151,17 @@ describe("parser", function () {
         expect(plusLexer.getNumber()).toEqual(205.88);
       });
 
-      it("should treat a single decimal point, or minus sign, as zero", function () {
-        const dotInput = new StringStream(".");
-        const dotLexer = new Lexer(dotInput);
-        expect(dotLexer.getNumber()).toEqual(0);
+      it("should treat a single decimal point, or minus/plus sign, as zero", function () {
+        const validNums = [".", "-", "+", "-.", "+.", "-\r\n.", "+\r\n."];
+        for (const number of validNums) {
+          const validInput = new StringStream(number);
+          const validLexer = new Lexer(validInput);
 
-        const minusInput = new StringStream("-");
-        const minusLexer = new Lexer(minusInput);
-        expect(minusLexer.getNumber()).toEqual(0);
+          expect(validLexer.getNumber()).toEqual(0);
+        }
 
-        const numbers = ["..", "-.", "+.", "-\r\n.", "+\r\n."];
-        for (const number of numbers) {
+        const invalidNums = ["..", ".-", ".+"];
```

Comment:
If "." is valid and is 0, I'd say that ".." is equivalent to 0 0.
From a lexer pov, I'd say that if we have a non-number char after the "." then we return a token 0 for this one and then the lexer creates a new token for the second point, hence"..123" will be equivalent to 0 0.123.

Reply:
Is that actually what Adobe Reader does in this case, since the `Lexer.getNumber` method tries to mimic its behavior?

Reply:
Given that this patch already feels a little risky, I do wonder if we perhaps want to limit the scope a little here?

Reply:
So in Acrobat, ".." and ".+" generate an error but ".-" is considered as 0.
The three expressions are ok in Chrome (the parsing doesn't break because of them).